nixos/modules/misc/man-db.nix

```diff
@@ -52,13 +52,11 @@ in
     environment.systemPackages = [ cfg.package ];
     environment.etc."man_db.conf".text =
       let
-        mandbForBuild = if pkgs.stdenv.hostPlatform == pkgs.stdenv.buildPlatform then
-                          cfg.package
-                        else
-                          pkgs.buildPackages.man-db;
-        manualCache = pkgs.runCommand "man-cache" { } ''
+        manualCache = pkgs.runCommand "man-cache" {
```

Comment:
For splicing to work correctly you need to put the man cache in a separate file and use callPackage to pull in man-db. However we cannot use callPackage here because `cfg.package` will also not work with splicing and always pull in the architecture that one tries to build for.

Reply:
If that wouldn't work we should fix it and make it work but according to @Artturin it should work.